client.go

```diff
@@ -0,0 +1,63 @@
+package nomadclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+type Nomad struct {
+	Client dphttp.Clienter
+	URL string
+	// Name needed for checker later.
+	// Name string
+}
+
+func NewClient(NomadEndpoint, NomadCACert string, NomadTLSSkipVerify bool) (*Nomad, error){
+
+	dpHTTPClient := *dphttp.DefaultClient
+
+	if strings.HasPrefix(NomadEndpoint, "https://") {
+		fmt.Print("in if https")
+		var tlsConfig *tls.Config
+		if NomadCACert != "" {
```

Comment:
I think you could probably remove the need for this final else block by having an early bail out check. Something like:

```
if NomadCACert == "" && !NomadTLSSkipVerify  {
    return nil, errors.New("invalid configuration with https but no CA cert or skip verification enabled")
}
```